tests/rados/test_pool_osd_recovery.py

```diff
@@ -40,312 +41,471 @@ def run(ceph_cluster, **kw) -> int:
+
+            log.info("Deployed new hosts and deployed OSDs on them")
+        except Exception as e:
+            log.error(f"Failed with exception: {e.__doc__}")
+            log.exception(e)
+            raise
+
+    def remove_offline_host(host_node_name: str):
+        # Removing an OSD host and checking status
+        rm_host = get_node_by_id(ceph_cluster, host_node_name)
+        log.info(f"Identified host : {rm_host.hostname} to be removed from the cluster")
+
+        # get list of osd_id on the host to be removed
+        rm_osd_list = rados_obj.collect_osd_daemon_ids(osd_node=rm_host)
+        log.info(f"list of osds on {rm_host.hostname}: {rm_osd_list}")
+        # Starting to drain an offline host.
+        log.info("Starting to drain an offline host")
+        cephadm.shell([f"ceph orch host drain {rm_host.hostname} --force"])
+        # Sleeping for 2 seconds for removal to have started
+        time.sleep(2)
+        log.debug(f"Started drain operation on node : {rm_host.hostname}")
+
+        status_cmd = "ceph orch osd rm status -f json"
+        end_time = datetime.datetime.now() + datetime.timedelta(seconds=7200)
+        flag = False
+        while end_time > datetime.datetime.now():
+            out, err = cephadm.shell([status_cmd])
+            try:
+                drain_ops = json.loads(out)
+                for entry in drain_ops:
+                    if entry["drain_done_at"] is None or entry["draining"]:
+                        log.debug(
+                            f"Drain operations are going on host {rm_host.hostname} \nOperations: {entry}"
+                        )
+                        raise Exception(
+                            f"drain process for OSD {entry['osd_id']} is still going on"
+                        )
+                    log.info(f"Drain operation completed for OSD {entry['osd_id']}")
+                log.info(f"Drain operations completed on host : {rm_host.hostname}")
+                flag = True
+                break
+            except json.JSONDecodeError:
+                log.info(f"Drain operations completed on host : {rm_host.hostname}")
+                flag = True
+                break
+            except Exception as error:
+                if end_time < datetime.datetime.now():
+                    log.error(f"Hit issue during drain operations: {error}")
+                    raise Exception(error)
+                log.info("Sleeping for 120 seconds and checking again....")
+                time.sleep(120)
+
+        if not flag:
+            log.error(
+                "Drain operation not completed on the cluster even after 7200 seconds"
+            )
+            raise
+        # remove the offline host so that drain operation gets completed
+        log.info(f"Removing host {rm_host.hostname} from the cluster")
+        cephadm.shell([f"ceph orch host rm {rm_host.hostname} --force --offline"])
+        time.sleep(5)
+        log.info(
```

Comment:
Can we do `ceph orch host ls ` and actually confirm that it's no longer listed?

Reply:
Added the check to confirm the decrease in host count